File: Data Structures & Algorithms/trapping-rain-water/submission-0.js
class Solution {
    /**
     * @param {number[]} height
     * @return {number}
     */
    trap(height) {
        const size = height.length;
        let water = 0;

        for(let i=0;i<size;i++){
            let left=0; 
            let right = 0;

            for(let l=0;l<=i;l++){
                left = Math.max(left,height[l]);
            }

            for(let r=i;r<size;r++){
                right = Math.max(right,height[r]);
            }

            water += Math.min(left,right)-height[i];
        }
        return water;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0;
        int right = heights.size()-1;

        int curr=0;
        int res =0;
        while(left<right){
            curr = min(heights[left],heights[right])*(right-left);
            res = max(curr,res);
            if(heights[left]<=heights[right]){
                left++;
            }else{
                right--;
            }
        }
        return res;
    }
};
